spectacles/validators.py

```diff
@@ -277,15 +277,16 @@ def _extract_error_details(query_result: dict) -> dict:
         if isinstance(data, dict):
             errors = data.get("errors") or [data.get("error")]
             first_error = errors[0]
-            message = first_error["message_details"]
-            if not isinstance(message, str):
-                raise TypeError(
-                    "Unexpected message type. Expected a str, "
-                    f"received type {type(message)}: {message}"
+            message = " ".join(
+                filter(
+                    None,
+                    (first_error.get("message"), first_error.get("message_details")),
                 )
+            )
```

Comment:
IMO, you could do this slightly more cleanly by using the second parameter of `get`, the default response if the key is not found.

```suggestion
message = " ".join(
    [first_error.get("message", ""), first_error.get("message_details", "")]
)
```